lumen/sources/duckdb.py

```diff
@@ -422,3 +416,53 @@ def _process_uri(self, tls=False, username=None, password=None):
         else:
             hostname = uri
         return protocol, hostname, username, password
+
+    def _execute_dremio_sql(self, sql_expr, as_pandas=False):
+        """
+        Executes a SQL expression on Dremio via Apache Arrow Flight.
+        """
+        from pyarrow import flight
+
+        flight_desc = flight.FlightDescriptor.for_command(sql_expr)
+        options = flight.FlightCallOptions(headers=self._headers)
+        flight_info = self._dremio_client.get_flight_info(flight_desc, options)
+        reader = self._dremio_client.do_get(flight_info.endpoints[0].ticket, options)
+        data_table = reader.read_all() if not as_pandas else reader.read_pandas()
+        return data_table
+
+    def _ingest_table(self, table):
+        """
+        Ingests a table from Dremio to DuckDB.
+        """
+        from pyarrow.dataset import dataset as arrow_dataset
+
+        sql_expr = self.get_sql_expr(table)
+        data_table = self._execute_dremio_sql(sql_expr)
+        arrow_ds = arrow_dataset(source=[data_table])
+        self._connection.from_arrow(arrow_ds).to_view(table.replace('"', ''))
```

Comment:
Not sure what the best way to handle the quotes. I think dremio needs them to read, but duckdb does not support it, especially in table names.

Reply:
Okay added quote/unquote methods and the pipeline seems to work well!